llvm/lib/Target/ARM/MCTargetDesc/ARMMCTargetDesc.cpp

```diff
@@ -431,6 +443,129 @@ class ARMMCInstrAnalysis : public MCInstrAnalysis {
   std::optional<uint64_t>
   evaluateMemoryOperandAddress(const MCInst &Inst, const MCSubtargetInfo *STI,
                                uint64_t Addr, uint64_t Size) const override;
+
+  std::vector<std::pair<uint64_t, uint64_t>>
+  findPltEntries(uint64_t PltSectionVA, ArrayRef<uint8_t> PltContents,
+                 const MCSubtargetInfo &STI) const override {
+    llvm::endianness DataEndianness = STI.getTargetTriple().isLittleEndian()
+                                          ? endianness::little
+                                          : endianness::big;
+    llvm::endianness InstrEndianness =
+        STI.checkFeatures("+big-endian-instructions") ? endianness::big
+                                                      : endianness::little;
+
+    // Do a lightweight parsing of PLT entries.
+    std::vector<std::pair<uint64_t, uint64_t>> Result;
+    if (STI.checkFeatures("+thumb-mode")) {
+      for (uint64_t Byte = 0, End = PltContents.size(); Byte + 12 < End;
+           Byte += 16) {
+        // Expected instruction sequence:
+        //
+        // movw ip, #lower16
+        // movt ip, #upper16
+        // add ip, pc
+        // ldr.w pc, [ip]
+        // b . -4
+
+        // Check for movw.
```

Comment:
With the additional comments above and variable names I think the // Check for <insn> comments aren't needed anymore.